troubleshooting.md

```diff
@@ -0,0 +1,211 @@
+
+When running this tutorial on a Fedora installation there is an issue that can prevent clusters from advancing past the `Provisioning` stage. 
+
+This problem can cause system instability. The root cause is a resource leak when Docker runs haproxy with the default Fedora configuration.
+
+To fix this issue you need to find the Docker systemd unit file. On most systems it should be at `/usr/lib/systemd/system/docker.service`. To find where it is on your system run:
+
+```bash
+systemctl status docker | grep loaded
+```
+Output:
+```bash
+     Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled; vendor preset: disabled)
+```
+Open this file with a text editor:
+
+e.g.
+```bash
+nano /usr/lib/systemd/system/docker.service
+```
+
+The file should have a section that looks like:
+```bash
+[Service]
+Type=notify
+# the default is not to use systemd for cgroups because the delegate issues still
+# exists and systemd currently does not support the cgroup feature set required
+# for containers run by docker
+ExecStart=/usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock
+ExecReload=/bin/kill -s HUP $MAINPID
+TimeoutSec=0
+RestartSec=2
+Restart=always
+```
+
+Modify it by adding the `--default-ulimit nofile=65883:65883` argument to the `ExecStart` line. The line should look similar to the below:
+
+```bash
+ExecStart=/usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock  --default-ulimit nofile=65883:65883
+```
+
+Write and exit the file, and reload the systemd config and D:ocker service using:
+```bash
+systemctl daemon-reload
+systemctl restart docker
+```
+
+You can check the status of the Docker service using:
+```bash
+systemctl status docker
+
+```
+
+The output on a successful reconfiguration should look like the below 
+
+```bash
+● docker.service - Docker Application Container Engine
+     Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled; vendor preset: disabled)
+     Active: active (running) since Wed 2022-10-19 14:31:51 BST; 13s ago
+TriggeredBy: ● docker.socket
+       Docs: https://docs.docker.com
+   Main PID: 13857 (dockerd)
+      Tasks: 49
+     Memory: 40.9M
+        CPU: 785ms
+     CGroup: /system.slice/docker.service
+             ├─ 13857 /usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock --default-ulimit nofile=65883:65883  #ulimit nofile limit is set
+             ├─ 14040 /usr/bin/docker-proxy -proto tcp -host-ip 127.0.0.1 -host-port 41825 -container-ip 172.19.0.2 -container-port 6443
+             ├─ 14066 /usr/bin/docker-proxy -proto tcp -host-ip 127.0.0.1 -host-port 38625 -container-ip 172.19.0.3 -container-port 6443
+             └─ 14143 /usr/bin/docker-proxy -proto tcp -host-ip 127.0.0.1 -host-port 5000 -container-ip 172.19.0.4 -container-port 5000
+```
+
+Once this change is working a Fedora system should be able to run the load balancer container correctly.
+
+## Too many open files
```

Comment:
Can we pro-actively link this section already where we describe the Docker setup for Linux?

Just that folks already verify during Docker installation if those values are fine